log/logger_test.go

```diff
@@ -14,6 +15,28 @@ const (
 	logName = "test"
 )
 
+func TestNewLoggerError(t *testing.T) {
+	// we expect an error from NewLoggerE in the event that we provide an
+	// unwriteable directory
+
+	targetDir := "/definitelyDoesNotExist"
+
+	// TODO(timraymond): this is some duplicated logic from
+	// (*Logger).getFileName. It should be possible to make it a publicly
+	// callable function to make this a little less brittle
+	fullLogPath := path.Join(targetDir, logName+".log")
+
+	// confirm the assumptions of this test before we run it:
+	if _, err := os.Stat(fullLogPath); err == nil {
+		t.Skipf("The log file at %q exists, so this test cannot sensibly run. Delete it first", fullLogPath)
```

Comment:
would it be better to use a relative path and always delete it during test setup?

Reply:
@rbtr I like the general idea. I did a bit more research and found [this SO post](https://unix.stackexchange.com/questions/643520/is-there-a-file-that-will-always-not-exist). The Go equivalent is [`os.TempDir`](https://pkg.go.dev/os@go1.17.6#TempDir) which also does the right thing on Linux and Windows, so I'm inclined to use it. Patch forthcoming :)

Reply:
+1 on `os.TempDir`